File: yavel/src/androidTest/java/com/awolity/yavel/YavelKeyTest.java
package com.awolity.yavel;

import androidx.test.runner.AndroidJUnit4;

import org.junit.After;
import org.junit.Assert;
import org.junit.Test;
import org.junit.runner.RunWith;

import javax.crypto.SecretKey;

import static org.junit.Assert.*;

@SuppressWarnings("unused")
@RunWith(AndroidJUnit4.class)
public class YavelKeyTest {

    @Test
    public void createKey_getKey_isCorrect() throws Exception {
        YavelKey yavelKey = new YavelKey("whatever");

        SecretKey sKey = yavelKey.getKey();
        assertNotNull(sKey);
    }

    @Test
    public void hasKey_isCorrect() throws Exception {
        YavelKey yavelKey = new YavelKey("whatever");
        Assert.assertTrue(yavelKey.hasKey());
    }

    @Test
    public void deleteKey_isCorrect() throws Exception {
        YavelKey yavelKey = new YavelKey("whatever");
        yavelKey.deleteKey();
        Assert.assertFalse(yavelKey.hasKey());
    }

    @After
    public void deleteKey() throws Exception{
        YavelKey yavelKey = new YavelKey("whatever");
        yavelKey.deleteKey();
    }
}

File: yavel/src/androidTest/java/com/awolity/yavel/YavelKeyTest.kt
package com.awolity.yavel

import androidx.test.runner.AndroidJUnit4

import org.junit.After
import org.junit.Assert
import org.junit.Test
import org.junit.runner.RunWith

import org.junit.Assert.*

@RunWith(AndroidJUnit4::class)
class YavelKeyTest {

    @Test
    @Throws(Exception::class)
    fun createKey_getKey_isCorrect() {
        val yavelKey = YavelKey("whatever")

        val sKey = yavelKey.key
        assertNotNull(sKey)
    }

    @Test
    @Throws(Exception::class)
    fun hasKey_isCorrect() {
        val yavelKey = YavelKey("whatever")
        Assert.assertTrue(yavelKey.hasKey())
    }

    @Test
    @Throws(Exception::class)
    fun deleteKey_isCorrect() {
        val yavelKey = YavelKey("whatever")
        yavelKey.deleteKey()
        Assert.assertFalse(yavelKey.hasKey())
    }

    @After
    @Throws(Exception::class)
    fun deleteKey() {
        val yavelKey = YavelKey("whatever")
        yavelKey.deleteKey()
    }
}
